libs/libarchfpga/src/physical_types.h

```diff
@@ -626,14 +610,52 @@ struct t_physical_tile_type {
 
     int index = -1; /* index of type descriptor in array (allows for index referencing) */
 
-    int logical_block_index = -1; /* index of the corresponding logical block type */
+    std::vector<std::string> equivalent_sites_names;
+    std::vector<t_logical_block_type_ptr> equivalent_sites;
+
+    /* Map holding the priority for which this logical block needs to be placed.
+     * logical_blocks_priority[priority] -> vector holding the logical_block indices */
+    std::map<int, std::vector<int>> logical_blocks_priority;
 
-    std::vector<t_equivalent_site> equivalent_sites;
+    /* Unordered map indexed by the logical block index.
+     * tile_block_pin_directs_map[logical block index][logical block pin] -> physical tile pin */
+    std::unordered_map<int, vtr::bimap<t_logical_pin, t_physical_pin>> tile_block_pin_directs_map;
 
     /* Returns the indices of pins that contain a clock for this physical logic block */
     std::vector<int> get_clock_pins_indices() const;
 };
-typedef const t_physical_tile_type* t_physical_tile_type_ptr;
+
+struct t_logical_pin {
+    int pin = -1;
+
+    t_logical_pin(int value) {
+        pin = value;
+    }
+
+    bool operator==(const t_logical_pin o) const {
+        return pin == o.pin;
+    }
+
+    bool operator<(const t_logical_pin o) const {
+        return pin < o.pin;
+    }
+};
+
+struct t_physical_pin {
+    int pin = -1;
+
+    t_physical_pin(int value) {
+        pin = value;
+    }
+
+    bool operator==(const t_physical_pin o) const {
+        return pin == o.pin;
+    }
+
+    bool operator<(const t_physical_pin o) const {
+        return pin < o.pin;
+    }
+};
```

Comment:
I needed to find a way to use the `vtr::bimap` in order to map physical/logical pins in case we need to access one having the other and viceversa. `vtr::bimap` though does not allow (as far as I have seen), having the same types when doing the mapping. Instead of modifying the bimap library I have preferred to go with the wrapping solution